### PR TITLE
Create `<SyncingModal />` component, along with a `<Dialog />` Penumbra UI component

### PR DESCRIPTION
**Pull request by jessepinho**
🚨 23 of the changed files are just renaming the Penumbra UI library's `<ThemeProvider />` component to the more aptly named `<PenumbraUIProvider />`.

https://github.com/user-attachments/assets/4b503949-be7f-4eec-a1ed-88ad805167a2

## In this PR
- Create a `<Dialog />` component in the Penumbra UI library
- Create a `<SyncingDialog />` component in minifront — **this is the first component in Minifront that uses the new Penumbra UI library!**
- Rename `<ThemeProvider />` to `<PenumbraUIProvider />`
- Add a few new values to theme (`blur` and `color.base`)
- Add `<PenumbraUIProvider />` to the root of minifront, so that minifront can successfully use Penumbra UI components
- Delete the `styled-components.d.ts` declaration merge we were previously using, and move it into the `theme.ts` file under `<PenumbraUIProvider />`. This was necessary because minifront didn't recognize the declaration merging we were doing, since minifront wasn't importing `packages/ui/src/styled-components.d.ts` (and we don't want to force consumers to import an ambient declaration file). Now that it's moved into `theme.ts`, the types automatically get merged

Closes https://github.com/penumbra-zone/web/issues/1377

### CODE REVIEW

File: packages/ui/src/Dialog/index.stories.tsx
Comment (jessepinho) [inline packages/ui/src/Dialog/index.stories.tsx:1]: https://penumbra-ui-preview--pr1584-jessepinho-card-vf8z1vua.web.app/?path=/docs/ui-library-dialog--docs

File: packages/ui/src/Text/index.tsx [hunk 1]
```diff
@@ -178,6 +203,19 @@ export type TextProps = TextType & {
...
+ */
+const omit = <ObjectType extends Record<string, unknown>>(
+  object: ObjectType,
+  key: keyof ObjectType,
+) => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars -- we're discarding the unused key
+  const { [key]: _, ...rest } = object;
+  return rest;
+};
+
```
Comment (jessepinho) [inline packages/ui/src/Text/index.tsx:218]: Created this to fix a previous issue where React was passing `h1`, `technical`, `large`, etc. as attributes to the DOM.

File: apps/minifront/src/components/layout.tsx [hunk 2]
```diff
@@ -3,21 +3,24 @@ import { HeadTag } from './metadata/head-tag';
 import { Header } from './header/header';
 import { Toaster } from '@repo/ui/components/ui/toaster';
 import { Footer } from './footer/footer';
-import { MotionConfig } from 'framer-motion';
 import '@repo/ui/styles/globals.css';
+import { SyncingModal } from './syncing-modal';
 
 export const Layout = () => {
   return (
-    <MotionConfig transition={{ duration: 0.1 }}>
```
Comment (jessepinho) [inline apps/minifront/src/components/layout.tsx:11]: This is already included in `PenumbraUIProvider`, which is now at the top of the component tree in `main.tsx`

File: packages/ui/src/ButtonGroup/index.tsx [hunk 3]
```diff
@@ -48,6 +50,12 @@ export interface ButtonGroupProps<IconOnly extends boolean> {
    * Will be used for all buttons in the group.
    */
   iconOnly?: IconOnly;
+  /**
+   * In some cases, you will want to disable responsiveness and enforce that
+   * buttons are always rendered in a column, rather than a row (such as in a
+   * narrow layout width, such as a dialog). In those cases, set this to `true`.
+   */
+  column?: boolean;
```
Comment (jessepinho) [inline packages/ui/src/ButtonGroup/index.tsx:58]: Added this for the purpose of the `Dialog` component, which will always display buttons in a column.

**Comment by jessepinho** [comment]
sorry all, I accidentally pushed up a commit to this branch that was supposed to be on a diff branch. Just fixed it

**Review by VanishMax — APPROVED** [review]
It's nice to see some new designs being applied. And a little sad that' after users will see the beautiful 'syncing' screen, it will close and show the old designs. 

Also, `Dialog` with Radix is extremely neat. Support for such UI library is going to be easy

**Pull request merged**